src/main/scala/wdl4s/Task.scala

```diff
@@ -30,6 +30,38 @@ object Task {
       case x: Ast => ParameterCommandPart(x, wdlSyntaxErrorFormatter)
     }
 
+    commandTemplate collect { case x: ParameterCommandPart => x } foreach { p =>
+      p.expression.ast.findTerminals("identifier") foreach {variable =>
+        if (!declarations.map(_.name).contains(variable.getSourceString)) {
+          throw new SyntaxError(wdlSyntaxErrorFormatter.commandExpressionContainsInvalidVariableReference(taskNameTerminal, variable))
+        }
+      }
+    }
+
+    def findDeclarationAsts(decl: Declaration) =
+      ast.findAsts(AstNodeName.Declaration) collect { case a: Ast if a.getAttribute("name").sourceString == decl.name => a }
+
+    declarations.foldLeft(Seq.empty[Declaration]) { (acc, cur) =>
+      findDeclarationAsts(cur) match {
+        case declAsts if declAsts.size > 1 =>
+          val declNameTerminals = declAsts.map(_.getAttribute("name").asInstanceOf[Terminal])
+          throw new SyntaxError(wdlSyntaxErrorFormatter.variableDeclaredMultipleTimes(declNameTerminals(0), declNameTerminals(1)))
+        case _ =>
+      }
+
+      cur.expression match {
+        case Some(e) => e.ast.findTerminals("identifier") foreach {variable =>
+          if (!acc.map(_.name).contains(variable.getSourceString)) {
+            // .head below because we are assuming if you have a Declaration object that it must have come from a Declaration AST
+            val declarationAst = findDeclarationAsts(cur).head
+            throw new SyntaxError(wdlSyntaxErrorFormatter.declarationContainsInvalidVariableReference(declarationAst.getAttribute("name").asInstanceOf[Terminal], variable))
+          }
+        }
+        case None =>
+      }
```

Comment:
converting the `match` to a `foreach` would eliminate the pattern match on `Some` and the need to have a `None` cass